internal/venice-common/src/main/java/com/linkedin/venice/system/store/MetaStoreReader.java

```diff
@@ -0,0 +1,71 @@
+package com.linkedin.venice.system.store;
+
+import static com.linkedin.venice.system.store.MetaStoreWriter.KEY_STRING_STORE_NAME;
+
+import com.linkedin.d2.balancer.D2Client;
+import com.linkedin.venice.client.store.AvroSpecificStoreClient;
+import com.linkedin.venice.client.store.ClientConfig;
+import com.linkedin.venice.client.store.ClientFactory;
+import com.linkedin.venice.common.VeniceSystemStoreUtils;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.systemstore.schemas.StoreMetaKey;
+import com.linkedin.venice.systemstore.schemas.StoreMetaValue;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.io.Closeable;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class MetaStoreReader implements Closeable {
+  private static final Logger LOGGER = LogManager.getLogger(MetaStoreReader.class);
+  private final Map<String, AvroSpecificStoreClient<StoreMetaKey, StoreMetaValue>> veniceClients =
+      new VeniceConcurrentHashMap<>();
+  private final D2Client d2Client;
+  private final String clusterDiscoveryD2ServiceName;
+
+  public MetaStoreReader(D2Client d2Client, String clusterDiscoveryD2ServiceName) {
+    this.d2Client = d2Client;
+    this.clusterDiscoveryD2ServiceName = clusterDiscoveryD2ServiceName;
+  }
+
+  public long getHeartbeat(String storeName) {
+    AvroSpecificStoreClient<StoreMetaKey, StoreMetaValue> client = getVeniceClient(storeName);
+    StoreMetaKey key =
+        MetaStoreDataType.HEARTBEAT.getStoreMetaKey(Collections.singletonMap(KEY_STRING_STORE_NAME, storeName));
+    try {
+      StoreMetaValue value = client.get(key).get();
```

Comment:
I think we need to specify a timeout here to avoid waiting forever.

Reply:
OK I copied the class from PushStatusStoreReader. 

It looks like all the usage in that class also don't have timeout, I am thinking to add a default 1s (that's even too much for this small data) as timeout. What do you think? This is a very specific usage, and I don't think it needs extra config for that.

Reply:
Let us use a higher timeout, such as 3 seconds since the first several requests can be slow because connection setup and warming.